kuma/wiki/search.py

```diff
@@ -63,6 +63,11 @@ def get_connection(cls, alias='default'):
     def get_doc_type(cls):
         return cls._doc_type.name
 
+    @classmethod
+    def case_insensitive_keywords(cls, keywords):
```

Comment:
Any reason to use lowercase keywords?

Reply:
You can use `simple` analyzer in the feild so Elasticsearch will use lowercase tokenizer.

Reply:
An analyzer is useful for  [Text](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/text.html) datatype, but these are [Keywords](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/keyword.html) and analyzers are not allowed. Using lowercase means that the same logic is used both when populating ElasticSearch and when querying it.

Reply:
If you want to have case insensitive search, then you must you `Text` datatype with simple analyzer instead of `Keywords` datatype. Storing different values in ES and database is not a good idea. If you store in lowercase, its impossible to get actual value anymore from Elasticsearch.

Reply:
The fields the ``case_insensitive_analyzer`` were used on are:

* ``css_classnames`` - extracted from the ``class`` elements in the content. This is a tool for front-end developers attempting to cross-reference CSS files to pages that use that CSS. I don't think our current front-end devs know this exists.
* ``html_attributes`` - extracted from the HTML elements in the content. This appears to be a tool for developers who are constructing the bleaching configuration. I haven't needed to do this kind of work in years.
* ``kumascript_macros`` - extracted from the content by looking for the ``{{macro}}`` pattern. KumaScript macros are case-insenstive at rendering time. This field gets used all the time, in the BCD migration project and macro retirement, and ambiguity in the queries has caused problems for these efforts.

It is simpler and more correct to use the ``Keyword`` datatype for these lists of keywords, where tokenization is already done in Python. The users is expecting a case-insensitive match when using these queries, so it makes sense to me to use Python lowercasing both when populating the index and querying, to make it easier to think about this parallel database.

We currently don't display what ElasticSearch stored for these queries, and I don't think we'll start, so there is no need to store exact values in ES for these keywords.